BLUESPAWN-client/headers/util/permissions/permissions.h

```diff
@@ -0,0 +1,284 @@
+		
+		/*Getter for the lpUserSID field*/
+		PSID GetUserSID() const;
+		/*Getter for the lpGroupSID field*/
+		PSID GetGroupSID() const;
+		/*Getter for the dacl field*/
+		PACL GetDACL() const;
+		/*Getter for the sacl field*/
+		PACL GetSACL() const;
+	};
+
+	/*Enum for storing type of Owner an Owner object is*/
+	enum OwnerType {
+		NONE, USER, GROUP 
+	};
+
+	class Owner : public Loggable {
+	protected:
+		//Whether or not this owner is on the system
+		bool bExists;
+
+		//The user's SID structure
+		SecurityDescriptor sdSID;
+
+		//Owner's qualified name
+		std::wstring wName;
+
+		//Domain to which the user belongs
+		std::wstring wDomainName;
+
+		//The type of the owner
+		OwnerType otType;
+
+	public:
+		/**
+		* Constructor for an owner object based off name
+		*
+		* @param name A wstring containing the name of an object. Other fields will
+		*	be filled in if an owner of that name exists.
+		*/
+		Owner(IN const std::wstring& name);
+		/**
+		* Constructor for an owner object based off sid
+		* 
+		* @param sid A SecurityDescriptor with lpUserSID set to the sid of the owner. Other
+		*	fields will be filled in if an owner of that sid exists. 
+		*/
+		Owner(IN const SecurityDescriptor& sid);
+		/**
+		* Constructor for an owner object that sets wName, bExists, and otOwnerType, but no other fields
+		*
+		* @param name A wstring containing value to be copied to wName
+		* @param exists A boolean containing value to be copied ot bExists
+		* @param t An OwnerType containing value to be copied to otOwnerType
+		*/
+		Owner(IN const std::wstring& name, IN const bool& exists, IN const OwnerType& t);
+		/**
+		* Constructor for an owner object that sets sdSID, bExists, and otOwnerType, but no other fields
+		*
+		* @param sid A SecurityDescriptor containing value to be copied to sdSID. Should have lpUserSID set 
+		*	to valid PSID if t is USER, and lpGroupSID set to valid PSID if t is GROUP.
+		* @param exists A boolean containing value to be copied ot bExists
+		* @param t An OwnerType containing value to be copied to otOwnerType
+		*/
+		Owner(IN const SecurityDescriptor& sid, IN const bool& exists, IN const OwnerType& t);
+		/**
+		* Constructor for an owner object that sets all fields to given values. Performs no checking
+		* that given name and sid line up. 
+		*
+		* @param name A wstring containing value to be copied to wName
+		* @ param domain A wstring containing value to be copied to wDomain
+		* @param sid A SecurityDescriptor containing value to be copied to sdSID. Should have lpUserSID set 
+		*	to valid PSID if t is USER, and lpGroupSID set to valid PSID if t is GROUP.
+		* @param exists A boolean containing value to be copied ot bExists
+		* @param t An OwnerType containing value to be copied to otOwnerType
+		*/
+		Owner(IN const std::wstring& name, IN const std::wstring& domain, IN const SecurityDescriptor& sid, IN const bool& exists, IN const OwnerType& t);
```

Comment:
This should be protected or private (users shouldn't be able to say whether or not it exists)

Reply:
BLUESPAWN must run with admin privs so this is not an issue?

Reply:
Not what I meant. Someone who doesn't understand this module may create an Owner object and accidentally set bExists to false. This would be to prevent that mistake. This is an inward facing method that is needed for internal functionality, but it's not something a user of this utility would need.